### PR TITLE
[MGDOBR-348] Setup dynamic context

### PR DESCRIPTION
**Pull request by radtriste**
[MGDOBR-348](https://issues.redhat.com/browse/MGDOBR-348) 

Issue splitted into different PRs for better review:
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/364
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/365
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/366
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/367
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/368

Depends on https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/366

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/BridgeSteps.java [hunk 1]
```diff
@@ -20,37 +21,41 @@
...
+        BridgeResource.getBridgeListFails(context.getManagerToken(), responseCode);
     }
 
     @When("create a Bridge with randomly generated name with access token")
     public void createRandomBridge() {
-        BridgeResponse response = BridgeResource.addBridge(BridgeUtils.retrieveAccessToken(),
-                StepsContext.randomBridgeName);
-        assertThat(response.getName()).isEqualTo(StepsContext.randomBridgeName);
+        String bridgeName = Utils.generateId("bridge");
+        while (isBridgeExisting(bridgeName)) {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/BridgeSteps.java:38]: In theory this can end in endless loop. Would be good to have limited tries to be sure.

Comment (radtriste) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/BridgeSteps.java]: see https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/367/commits/4e60f40dde1a67e05e60051b509c961c244389f6

File: integration-tests/src/test/resources/features/end2endtestit.feature [hunk 2]
```diff
@@ -3,9 +3,8 @@ Feature: End to End Bridge integration tests
   Scenario:By default Manager url should not be accessible without authentication
     Given get list of Bridge instances fails with HTTP response code 401
 
-
   Scenario: Bridge is created and in ready state
-    Given get list of Bridge instances with access token doesn't contain randomly generated Bridge
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/end2endtestit.feature:8]: Why did you delete this step?

Comment (radtriste) [reply integration-tests/src/test/resources/features/end2endtestit.feature:8]: the number generation is done in `create a Bridge with randomly generated name with access token`

Comment (sutaakar) [reply integration-tests/src/test/resources/features/end2endtestit.feature:8]: ok

**Review by sutaakar — APPROVED** [review]

**Review by Srihari1192 — APPROVED** [review]

**Review by jiripetrlik — APPROVED** [review]
